### PR TITLE
Improvements for employee table / customer list

### PR DESCRIPTION
**Pull request by thoasc**
This PR fixes a number of issues/bugs related to the employee table and customer list.

* [#570](https://github.com/knowit/Dataplattform-issues/issues/570): the index used when mapping competence and motivation columns in the filtering was off by one.
  * Also discovered and fixed that we did the aggregation for these scores twice per employee in the backend. Should improve response times a bit.
  * Added types for the employee table data, both in backend and frontend. This had some ripple effects into the customer list that required some refactoring, and I fixed the following issues at the same time:
* [#616](https://github.com/knowit/Dataplattform-issues/issues/616): the customer list now creates a separate copy of the employee table data, not mutating the original data.
* [#590](https://github.com/knowit/Dataplattform-issues/issues/590): when transforming the employee data, the customer list filters out employees without customer.

Also made some other adjustments:
* Fixed a problem where the employee table would not render correctly if the user sorted the table while a row was expanded.
* Made the `FallbackMessage` to a common component, as it is now used across pages.
* Made some changes to simplify the employee table data structure:
  * Removed unused/redundant fields (`guid`, `email_id`)
  * Changed the `rowId` to the employee email. This was previously an UUID that was regenerated in the backend on each request, making the id unstable/unreliable. Also removed the `uuid` dependency, as it was no longer used.
  * Moved the generation of `competenceUrl` from backend to frontend, as this is how it's done everywhere else. (Used for fetching the data for the expandable employee info in the table.)
* Created a common function in backend for generating CV download links
* Did some general cleanup in `data.ts` (backend)

I've tried to transfer all changes made in the backend to v2 of the API.

### CODE REVIEW

**Review by aspleym — COMMENTED** [review]
Kun små forslag

File: backend/routers/employees/employeesTypes.ts [hunk 1]
```diff
@@ -1,4 +1,25 @@
...
+      email: string
+      user_id: string
+      degree?: string
+    },
+    jobTitle: string | undefined,
+    projectStatus: ProjectStatus,
+    customer: Customer | undefined,
+    cvLinks: CvLinks | undefined,
+    motivationScores: Record<string, number>,
+    competenceScores: Record<string, number>
```
[the remark this answers is not in the document]
Comment (thoasc) [reply backend/routers/employees/employeesTypes.ts:19]: Skulle gjerne satt disse som optional, men typen for `rowData` er litt annerledes fordi dette ikke er felter i et objekt, men elementer i et array (se linje 6: `rowData` er et array). (Navnene på elementene her er bare for dokumentasjonens skyld, og altså ikke nøkler i et objekt.)

File: backend/routers/employees/aggregationHelpers.ts [hunk 2]
```diff
@@ -1,28 +1,22 @@
...
-export const getStorageUrl = (key: string) => {
-  if (key !== undefined) {
-    return `${process.env.STORAGE_URL}/${key}`
-  } else {
-    return undefined
+export const getStorageUrl = (key?: string) => {
+  if (!key) {
+    return
   }
+  return `${process.env.STORAGE_URL}/${key}`
```
[the remark this answers is not in the document]
Comment (thoasc) [reply backend/routers/employees/aggregationHelpers.ts:19]: Det er et godt spørsmål, som jeg tror kanskje blir en smakssak?

Fordelen med å tillate at disse funksjonene kan ta imot og returnere `undefined` er jo at man slipper å skrive den if-sjekken hver gang man kaller på funksjonen. I praksis vil jo ikke funksjonskallet ha noen effekt hvis verdien er `undefined`.

Men på den annen side blir det kanskje tydeligere i koden at man kan ende opp med `undefined` som resultat-verdi dersom man eksplisitt sjekker det ved funksjonskallene.

Så jeg er egentlig litt usikker 🤔

File: backend/routers/employees/employeesTypes.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (aspleym) [reply backend/routers/employees/employeesTypes.ts:19]: Sant det, det går ikke i array nei, trodde det var et objekt.

[the remark this answers is not in the document]
Comment (thoasc) [reply backend/routers/employees/employeesTypes.ts:19]: Jeg var ikke klar over det før jeg prøvde det her 😛

**Comment by thoasc** [comment]
Ser nå at denne PR-en ble litt stor, så lukker den og splitter opp i mindre deler.

**Pull request closed without merging**